plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPageSink.java

```diff
@@ -112,8 +113,12 @@ public JdbcPageSink(ConnectorSession session, JdbcOutputTableHandle handle, Jdbc
 
         String insertSql = jdbcClient.buildInsertSql(handle, columnWriters);
         try {
+            insertSql = remoteQueryModifier.apply(session, insertSql);
             statement = connection.prepareStatement(insertSql);
         }
+        catch (TrinoException e) {
+            throw closeAllSuppress(e, connection);
```

Comment:
I would throw new `TrinoException` with `e` as cause. So all you need to do is `catch (SQLException | TrinoException e)`